rails-snow-queen/app/assets/javascripts/map.js

```diff
@@ -0,0 +1,110 @@
+class Map {
+	constructor(config) {
+    this.map = this.initMap();
+    this.drawingManager = this.initDrawingManager();
+    this.removeControl = this.initRemoveControl();
+    this.geocoder = this.initGeocoder();
+    this.marker = null;
+    this.geocodedAddress = null;
+    this.polygons = [];
+    this.onGeocodingResponse = config.onGeocodingResponse;
+    this.onPolygonsCreated = config.onPolygonsCreated;
+
+    this.handleSearchSubmit = this.handleSearchSubmit.bind(this);
+    this.handleGeocodingResponse = this.handleGeocodingResponse.bind(this);
+    this.handlePolygonCreated = this.handlePolygonCreated.bind(this);
+    this.handleRemovePolygon = this.handleRemovePolygon.bind(this);
+    this.addListeners()
+  }
+  
+  initMap() {
+    return new google.maps.Map(document.getElementById('map'), {
+      zoom: 14,
+      center: {lat: 49.2860, lng: -122.8130},
+      mapTypeId: google.maps.MapTypeId.SATELLITE,
+      tilt:0
+    });
+  }
+  
+  initDrawingManager() {
+    const drawingOptions = {
+      fillColor: '#2DC1D6',
+      fillOpacity: 0.3,
+      strokeWeight: 5,
+      strokeColor: '#2DC1D6',
+      clickable: false,
+      // editable: true,
+      zIndex: 1
+    }
+    return new google.maps.drawing.DrawingManager({
+      drawingMode: google.maps.drawing.OverlayType.POLYGON,
+      drawingControl: true,
+      drawingControlOptions: {
+        position: google.maps.ControlPosition.TOP_CENTER,
+        drawingModes: ['polygon', 'rectangle']
+      },
+      map: this.map,
+      // markerOptions: {icon: 'https://developers.google.com/maps/documentation/javascript/examples/full/images/beachflag.png'},
+      // markerOptions: {icon: 'snow_flake.png'},
+      polygonOptions: drawingOptions,
+      rectangleOptions: drawingOptions
+    })
+  }
+  
+  initRemoveControl() {
+    const removeControlDiv = document.createElement('div');
+    removeControlDiv.classList.add("map-btn")
+    removeControlDiv.title = 'Click to remove selected area from the map';
+    removeControlDiv.innerHTML = 'Remove last';
+
+    removeControlDiv.index = 1;
+    this.map.controls[google.maps.ControlPosition.TOP_CENTER].push(removeControlDiv);
+
+    return removeControlDiv;
+  }
+
+  initGeocoder() {
+    return new google.maps.Geocoder();
+  }
+
+  addListeners() {
+    document.getElementById('AddressSearch').addEventListener('submit', this.handleSearchSubmit);
+    this.removeControl.addEventListener('click', this.handleRemovePolygon);
+    this.drawingManager.addListener('polygoncomplete', this.handlePolygonCreated)
+  }
+  
+  handleSearchSubmit(event) {
+    event.preventDefault();
+    if (this.marker) { this.marker.setMap(null) }
+    this.geocodeAddress();
+  }
+
+  geocodeAddress() {
+    let address = document.getElementById('address').value;
+    this.geocoder.geocode({'address': address, 'region': "CA"}, this.handleGeocodingResponse);
```

Comment:
Try changing `region: 'CA'` to `region: 'ca'` to see if it works better.

Reply:
If I understand correctly, the [documentation](https://developers.google.com/maps/documentation/javascript/geocoding#GeocodingRegionCodes) sets the region influence by default to the domain values:

> region — The region code, specified as a IANA language region subtag. In most cases, these tags map directly to familiar ccTLD ("top-level domain") two-character values. The region parameter will only influence, not fully restrict, results from the geocoder. See more information about region code biasing below.

Either case, the [IANA](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry) code is 'CA'

> %%
Type: region
Subtag: CA
Description: Canada
Added: 2005-10-16
%%

Reply:
I tried the change and it doesnt seem to make a difference, still won't default to canada

Reply:
You can also try adding a [`bounds` property to the Geocode Request](https://developers.google.com/maps/documentation/javascript/geocoding#GeocodingViewports) (in addition to the `region`):
> `bounds` — The `LatLngBounds` within which to bias geocode results more prominently. The bounds parameter will only influence, not fully restrict, results from the geocoder. See more information about viewport biasing below.

Reply:
I have added it to the task list to tackle later